### PR TITLE
[V2] Remove pyaes

### PR DESCRIPTION
**Pull request by Jahongir-Qurbonov**

### CODE REVIEW

**Comment by Jahongir-Qurbonov** [comment]
I will remove the draft status if [this](https://github.com/cher-nov/cryptg/pull/36) is merged

**Comment by Lonami** [comment]
> I will remove the draft status if [this](https://github.com/cher-nov/cryptg/pull/36) is merged

Does it also need a new release?

**Comment by Jahongir-Qurbonov** [comment]
> Does it also need a new release?

Yes

**Comment by Lonami** [comment]
Won't be able to look into it for a few days but I'll make a note.

**Comment by Lonami** [comment]
[cryptg 0.5](https://pypi.org/project/cryptg/0.5.0/) has been published to PyPi, and just like last time, I've done it locally, same for the binaries built on CI. So let me know if something's missing.

Seeing how [Build wheels on aarch64 for ubuntu-20.04](https://github.com/cher-nov/cryptg/actions/runs/11276736534/job/31361254043) failed with a timeout after 6h, I've decided to cancel this build on the 0.5 tag as well, as it's just a waste of resources. If that's needed we likely need to take a closer look.

**Comment by cher-nov** [comment]
> [cryptg 0.5](https://pypi.org/project/cryptg/0.5.0/) has been published to PyPi

Could you please also make a corresponding release tag there in repository?

**Comment by Lonami** [comment]
The tag should've been pushed already, but I've created a release now https://github.com/cher-nov/cryptg/releases/tag/v0.5.

File: client/pyproject.toml [hunk 1]
```diff
@@ -19,15 +19,14 @@ classifiers = [
     "Typing :: Typed",
 ]
 dependencies = [
-  "pyaes~=1.6",
```
Comment (Lonami) [inline client/pyproject.toml]: Being able to run Telethon with Python-only dependencies was a very nice feature. I do not want to see it go.

Can we at least make `pyaes` optional for people who can't or don't want to run `cryptg`? Using `cryptg` as a default is probably a good idea, but removing `pyaes` support is unfortunate.

Not sure how to make this work if `cryptg` is currently required though.

**Review by Lonami — CHANGES_REQUESTED** [review]

File: client/pyproject.toml (hunk 1 above)
Comment (Jahongir-Qurbonov) [reply client/pyproject.toml]: Understandable, I'll fix it when I have time. So should cryptg be optional or default?

Comment (Lonami) [reply client/pyproject.toml]: Optional but default should be fine. Telethon has come a long way and CI can build the dependency for a lot of targets, so *most* of the time I'd expect it to work. But for weird or source-only environments, falling back to pyaes is better than nothing.

**Pull request closed without merging**